In [ ]:
require(readr)


Loading required package: readr



In [ ]:
install.packages("car")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

also installing the dependencies ‘numDeriv’, ‘SparseM’, ‘MatrixModels’, ‘minqa’, ‘nloptr’, ‘Rcpp’, ‘RcppEigen’, ‘carData’, ‘abind’, ‘pbkrtest’, ‘quantreg’, ‘lme4’




In [ ]:
install.packages("lmtest")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

also installing the dependency ‘zoo’




In [ ]:
library(lmtest)
library(car)

Loading required package: zoo


Attaching package: ‘zoo’


The following objects are masked from ‘package:base’:

    as.Date, as.Date.numeric


Loading required package: carData



In [ ]:
install.packages("DescTools")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

also installing the dependencies ‘proxy’, ‘rootSolve’, ‘e1071’, ‘lmom’, ‘mvtnorm’, ‘expm’, ‘Exact’, ‘gld’, ‘BH’




In [ ]:
library(DescTools)


Attaching package: ‘DescTools’


The following object is masked from ‘package:car’:

    Recode




In [ ]:
consumo=read.csv("/content/cc_consumo.csv")

# Regresión logística consumo de cigarrillo

In [ ]:
consumo=consumo[,-9]
consumo=consumo[,-1]
head(consumo)

id   educacion sexo      edad cervezas hijos cigarrillo
1 1003 16        Masculino 41   1        0     1         
2  522  9        Masculino 33   5        1     1         
3 1086 17        Masculino 51   3        4     0         
4  659  8        Masculino 46   6        2     0         
5  771 12        Masculino 18   3        0     0         
6  331 12        Femenino  57   4        3     1

In [ ]:
consumo$sexo=as.factor(consumo$sexo)

In [ ]:
rlogistica=glm(cigarrillo~educacion+sexo+edad+cervezas+hijos,data=consumo,family=binomial)
summary(rlogistica)


Call:
glm(formula = cigarrillo ~ educacion + sexo + edad + cervezas + 
    hijos, family = binomial, data = consumo)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.1968  -0.9111  -0.6754   1.0891   2.0417  

Coefficients:
              Estimate Std. Error z value Pr(>|z|)    
(Intercept)   -4.26143    0.63294  -6.733 1.66e-11 ***
educacion      0.08941    0.03367   2.656  0.00791 ** 
sexoMasculino  0.58074    0.16710   3.475  0.00051 ***
edad           0.06107    0.00728   8.390  < 2e-16 ***
cervezas       0.03839    0.05046   0.761  0.44686    
hijos         -0.07258    0.07343  -0.988  0.32294    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 987.99  on 729  degrees of freedom
Residual deviance: 879.71  on 724  degrees of freedom
AIC: 891.71

Number of Fisher Scoring iterations: 4


De acuerdo a la prueba de significancia individual para la regresion logística (prueba Wald, Pr(>|z|) las variables cervezas e hijos tienen un p-valor cada una, mayor a alfa (5%) por lo que no se rechaza la hipotesis nula, es decir,  no aportan al modelo. Por esta razón se replantea el modelo abajo.

In [ ]:
rlogistica2=glm(cigarrillo~educacion+sexo+edad,data=consumo,family=binomial)
summary(rlogistica2)


Call:
glm(formula = cigarrillo ~ educacion + sexo + edad, family = binomial, 
    data = consumo)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.1616  -0.9061  -0.6557   1.0969   2.0546  

Coefficients:
               Estimate Std. Error z value Pr(>|z|)    
(Intercept)   -4.148205   0.559507  -7.414 1.23e-13 ***
educacion      0.089218   0.031181   2.861 0.004219 ** 
sexoMasculino  0.611335   0.163859   3.731 0.000191 ***
edad           0.057681   0.006395   9.020  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 987.99  on 729  degrees of freedom
Residual deviance: 881.25  on 726  degrees of freedom
AIC: 889.25

Number of Fisher Scoring iterations: 4


In [ ]:
odds=exp(coef(rlogistica2))
odds

(Intercept)     educacion sexoMasculino          edad 
   0.01579273    1.09331951    1.84289035    1.05937693

De acuerdo a los datos anteriores por cada año de educación que tengan las personas aumenta la probabilidad de consumir cigarrillo  9.33%. Además dado el caso que sea hombre la probabilidad de consumir cigarrillo aumenta el 84.28%. Por último, cada por cada año de edad de la persona, la probabilidad de consumir cigarrillo aumenta 5.93%.

In [ ]:
consumo$Pr_C=predict(rlogistica2,type="response")
head(consumo)

id   educacion sexo      edad cervezas hijos cigarrillo Pr_C     
1 1003 16        Masculino 41   1        0     1          0.5635405
2  522  9        Masculino 33   5        1     1          0.3035535
3 1086 17        Masculino 51   3        4     0          0.7153622
4  659  8        Masculino 46   6        2     0          0.4576518
5  771 12        Masculino 18   3        0     0          0.1934118
6  331 12        Femenino  57   4        3     1          0.5523646

In [ ]:
consumo$pronostico=as.numeric(consumo$Pr_C>0.5)
head(consumo)

id   educacion sexo      edad cervezas hijos cigarrillo Pr_C      pronostico
1 1003 16        Masculino 41   1        0     1          0.5635405 1         
2  522  9        Masculino 33   5        1     1          0.3035535 0         
3 1086 17        Masculino 51   3        4     0          0.7153622 1         
4  659  8        Masculino 46   6        2     0          0.4576518 0         
5  771 12        Masculino 18   3        0     0          0.1934118 0         
6  331 12        Femenino  57   4        3     1          0.5523646 1

In [ ]:
matriz_Confusion=table(consumo$cigarrillo,consumo$pronostico)
num=matriz_Confusion[1,1]+matriz_Confusion[2,2]
total=sum(matriz_Confusion)
matriz_Confusion=prop.table(matriz_Confusion,2)
matriz_Confusion



   
            0         1
  0 0.7131313 0.3319149
  1 0.2868687 0.6680851

In [ ]:
num/total

[1] 0.6986301

In [ ]:
1-(num/total)

[1] 0.3013699

Respecto a la bondad de ajuste, de acuerdo con los datos presentados por la matriz de confusión y que los valores son mayores al 50% cada, se puede afirmar que quienes efectivamente no fuman son el 71.31% y respecto a las personas que fuman es del 66.80%. Ahora bien calculando la tasa de acierto de la matriz de confusión, obtenemos que la tasa de acierto es del 69.86%

In [ ]:
PseudoR2(rlogistica2,which = "McFadden")
PseudoR2(rlogistica2,which = "Nagelkerke")

McFadden 
0.1080464

Nagelkerke 
 0.1834338

In [ ]:
anova(rlogistica2,test="Chisq")

Df Deviance  Resid. Df Resid. Dev Pr(>Chi)    
NULL      NA        NA 729       987.9946             NA
educacion  1  0.783045 728       987.2115   3.762115e-01
sexo       1  9.961169 727       977.2504   1.598765e-03
edad       1 96.005010 726       881.2454   1.145931e-22

Respecto a la significancia global, de acuerdo a los datos suministrados en la columna Pr(>Chi), ya que los valores  en las filas educación, sexo y edad son menores que alfa, entonces se  rechaza la hipotesis nula. Por lo cual se puede afirmar que el modelo esta correctamente especificado.

In [ ]:
residuos=rstandard(rlogistica2)
summary(residuos)

    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
-2.17171 -0.90808 -0.65753 -0.05477  1.09965  2.05887 

In [ ]:
shapiro.test(residuos)


	Shapiro-Wilk normality test

data:  residuos
W = 0.85829, p-value < 2.2e-16


Respecto al supuesto de normalidad residual, p-valor es menor a alfa por lo que no existe normalidad residual.

In [ ]:
vif(rlogistica2)

educacion      sexo      edad 
 1.063259  1.014540  1.078222

Respecto al supuesto de no multicolinealidad, el factor de inflación de varianza es menor a 5 para todas las variables. Esto indica que hay independencia entre las variables educación, sexo y edad.

In [ ]:
lmtest::bptest(rlogistica2)


	studentized Breusch-Pagan test

data:  rlogistica2
BP = 14.255, df = 3, p-value = 0.002578


En el supuesto de homocedasticidad p-valor es menor que alfa por lo que se rechaza la hipótesis nula, es decir, los betas pueden ser sesgados y no confiables.

# Regresión lineal consumo cerveza

In [ ]:
head(consumo)

id   educacion sexo      edad cervezas hijos cigarrillo Pr_C      pronostico
1 1003 16        Masculino 41   1        0     1          0.5635405 1         
2  522  9        Masculino 33   5        1     1          0.3035535 0         
3 1086 17        Masculino 51   3        4     0          0.7153622 1         
4  659  8        Masculino 46   6        2     0          0.4576518 0         
5  771 12        Masculino 18   3        0     0          0.1934118 0         
6  331 12        Femenino  57   4        3     1          0.5523646 1

In [ ]:
rlineal=lm(cervezas~educacion+sexo+edad+hijos+cigarrillo,data=consumo)
summary(rlineal)


Call:
lm(formula = cervezas ~ educacion + sexo + edad + hijos + cigarrillo, 
    data = consumo)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.0917 -1.1287 -0.1657  0.9492  8.8099 

Coefficients:
               Estimate Std. Error t value Pr(>|t|)    
(Intercept)    5.589425   0.403999  13.835   <2e-16 ***
educacion     -0.206684   0.023988  -8.616   <2e-16 ***
sexoMasculino -0.137286   0.126542  -1.085    0.278    
edad          -0.004672   0.005232  -0.893    0.372    
hijos          0.014195   0.053956   0.263    0.793    
cigarrillo     0.108253   0.133795   0.809    0.419    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.648 on 724 degrees of freedom
Multiple R-squared:  0.1011,	Adjusted R-squared:  0.09491 
F-statistic: 16.29 on 5 and 724 DF,  p-value: 3.026e-15


Revisando la bondad de ajuste del modelo lineal para el consumo de cervezas tenemos que el porcentaje de consumo de cerveza que es explicado por Xi es del 10.11%, entonces no se cumple con el porcentaje admisible para el modelo.

En el diagnóstico de significancia individual tenemos que de acuerdo a la prueba t, la variable que aporta al modelo es educación ya que su  Pr(>|t|) es menor que alfa. Las otras variables tienen un  Pr(>|t|) mayor que alfa, por lo que no rechazan la hipótesis nula, es decir, no aportan al modelo.

En el diagnóstico de significancia global en la prueba t p-valor es menor que alfa, por lo que rechaza la hipotesis nula, es decir el modelo estaria correctamente especificado

De acuerdo a lo anterior intentando mejorar el modelo aplicaremos una segunda regresión en la que empleemos unicamente la variable que aporta al modelo, es decir, educación.


In [ ]:
rlineal2=lm(cervezas~educacion,data=consumo)
summary(rlineal2)


Call:
lm(formula = cervezas ~ educacion, data = consumo)

Residuals:
    Min      1Q  Median      3Q     Max 
-2.9419 -1.1108 -0.1281  0.8892  8.8719 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  5.36607    0.31060  17.277   <2e-16 ***
educacion   -0.20345    0.02282  -8.916   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.646 on 728 degrees of freedom
Multiple R-squared:  0.09844,	Adjusted R-squared:  0.0972 
F-statistic: 79.49 on 1 and 728 DF,  p-value: < 2.2e-16


Revisando la bondad de ajuste del modelo lineal para el consumo de cervezas tenemos que el porcentaje de consumo de cerveza que es explicado por Xi es del 9.84%, por lo cual el porcentaje restante sería explicado por el error. Con base en lo anterior, se evidencia que el modelo no tiene aún un porcentaje admisible de ajuste ya que debe ser superior al 50%

En el diagnóstico de significancia individual tenemos que de acuerdo a la prueba t, la variable que aporta al modelo es educación ya que su  Pr(>|t|) es menor que alfa. Las otras variables tienen un  Pr(>|t|) mayor que alfa, por lo que no rechazan la hipótesis nula, es decir, no aportan al modelo.

En el diagnóstico de significancia global en la prueba t p-valor es menor que alfa, por lo que rechaza la hipotesis nula, es decir el modelo estaria correctamente especificado

In [ ]:
residuos=rstandard(rlineal2)
shapiro.test(residuos)


	Shapiro-Wilk normality test

data:  residuos
W = 0.92647, p-value < 2.2e-16


En el supuesto  de normalidad residual el resultado nos arroja un p-valor menor que alfa,entoncesse rechaza la hipotesis nula, por lo cual no hay normalidad residual, es decir hay valores atípicos.

In [ ]:
vif(rlineal2)

ERROR: ignored

El supuesto de No multicolinealidad no aplica en este caso ya que solo quedaría la variable educación.

In [ ]:
lmtest::bptest(rlineal2)


En el supuesto de homocedasticidad de acuerdo a la prueba de Breusch Pagan, el p-valor es menor que alfa, entonces se rechazaría la hipotesis nula, por lo cual, habria heterocedasticidad, es decir, los valores de beta no serían confiables

# Preguntas

1.	¿Cuál es la variable que tiene mayor incidencia en el consumo de cigarrillo? ¿Cómo varía la probabilidad de consumir cigarrillo ante la presencia de los factores más determinantes?

Con base en la prueba Wald de significancia individual la variable que más aporta en el modelo es la edad, ya que su p-valor es el menor de todos lo valores con respecto a las otras variables pertenecieentes al modelo. Por esto la edad influiría en el consumo de cigarrillo.

Rspecto a la pregunta de como varia la probabilidad ante la presencia de los factores mas determinantes: la variable sexomasculino indica que dado la situación que el sujeto en cuestión sea hombre su probabilidad de consumir cigarrillo aumenta el 84.28% , respecto a la variable edad  cada año adicional del sujeto aumenta la probabilidad de consumir el 5.93% y por último, un año adicional de educación aumenta la probabilidad de consumir cigarrillo el 9.33%.


2.	¿Es confiable pronosticar la probabilidad de consumo de cigarrillo a partir de la información recolectada? ¿Qué situaciones correctivas deberían aplicarse para mejorar estos pronósticos?

Con base en los diagnóstico realizados (matriz de confusión, pseudo R^{2}, prueba de wald y chi cuadrado) se determino que el  modelo es confiable para pronósticar, ya que por una parte su tasa de acierto es del 69.86% superando el requerido del 50%, de acuerdo a la prueba de Wald se dejaron a las variables que aportaron al modelo y en el diagnóstico de significancia individual con chisq se determino que el modelo esta corectamente especificado. Sin embargo, se podrían aplica algunas correcciones del lado de los supuestos para mejorar los pronósticos como por ejemplo el uso de minimos cuadrados perfectos para igualar la dispersión de los residuos.


3.	¿Podría postular otros factores que potencialmente ayudaran a explicar el consumo de cigarrillo?

Se podrian tener en cuenta factores  como la exposición a publicidad entorno al consumo de cigarrillo, ya que la estrategia de marketing de una emrpeesa puede influir en la elección y compra de un producto en este caso del cigarrillo; estar en un entorno con personas fumadoras ya que la persona puede sentir el impulso a fumar como una forma de conocer e interactuar con otros, o una manera de conseguir la aceptación de grupo; finalmente, el estado de animo de las personas ya que personas en situaciones depresivas pueden ser influenciados a fumar. Posterior a tener una conjunto de datos recolectados se realizaría un diagnóstico para determinar su relevancia en el pronóstico.

4.	¿Cuáles son los factores que inciden en la explicación del consumo de cerveza por parte de la población encuestada? ¿En qué porcentaje estos factores explican la variabilidad del consumo de cerveza?

De acuerdo al modelo  de regresión lineal y a partir de la prueba t en la significancia individual, el factor que incide en el consumo de cerveza es el nivel de educación.

Xi en este caso los años de educación de acuerdo a la prueba R^{2} explica el 9.84% de la variabilidad de y es decir, el  consumo de cerveza. Esto nos dice que al modelo le hacen falta factores que puedan explicar en una mejor medida el consumo de cerveza, ya que para este caso el error restante que terinaria explicando a la variable dependiente seria 1-R^{2} que corresponde aproximadamente con un 90.16%.

5.	¿Es confiable realizar pronósticos o análisis estructurales con los factores obtenidos en el modelo del punto anterior? Explique detalladamente su respuesta

No es confiable realizar pronósticos con el factor educación ya que de acuerdo al diagnóstico de bondad de ajuste, en la prueba se determinó que la variable aporta únicamente un 9.84% a la explicación del consumo de cigarrillo, quedando fuera del porcentaje admisible (que debe ser mayor al 50%). Además que el porcentaje restante quedaría atribuido al término de error del modelo. En este sentido, se puede añadir que al intentar comprobar el supuesto de homocedasticidad con la prueba Breush Pagan, se rechazó hipótesis nula de homocedasticidad, por lo que el valor de beta sería sesgado y no confiable.

6.	¿Qué oportunidades de mejora y/o correctivos deben aplicarse para que se pueda explicar el consumo de cerveza con mayor precisión y confiabilidad?

Se deben utlizar factores que aporten significativamente al modelo, de manera que se logre una bondad de ajuste adecuada (R^{2} mayor al 50%). Para asegurar pronósticos ajustados se deberian eliminar lo valores atípicos en la variable de consumo de cerveza, de esta forma, a través de a preuba Shapiro se podría obtener un p-valor que siendo mayor que alfa arrojaría normalidad residual y por consiguiente, habrían resultados mas precisos con el modelo. Finalmente, para garantizar que los betas del modelo sean insesgados y confiables, se necesita que independiente de los valores de Xi, los valores residuales sean concentrados aplicando estimados de minimos cuadrados generalizados.
Con los elmentos enunciados en el parrafo anterior se podría mejorar el modelo permitiendo realizar un pronóstico adecuado del consumo de cerveza.